modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GoCliClientCodegen.java

```diff
@@ -0,0 +1,127 @@
+import java.util.Map;
+
+public class GoCliClientCodegen extends PureCloudGoClientCodegen {
+    protected Logger LOGGER = LoggerFactory.getLogger(GoCliClientCodegen.class);
+
+    public GoCliClientCodegen() {
+        super();
+
+        typeMapping.put("integer", "int");
+        typeMapping.put("long", "int");
+
+        outputFolder = "generated-code/go";
+
+        embeddedTemplateDir = templateDir = "go";
+    }
+
+    @Override
+    public String getName() {
+        return "clisdkclient";
+    }
+
+    @Override
+    public void processOpts() {
+        super.processOpts();
+
+        // Wipe out all the templates so we can start with a clean slate
+        this.apiTemplateFiles.clear();
+        this.apiDocTemplateFiles.clear();
+        this.apiTestTemplateFiles.clear();
+        this.modelTemplateFiles.clear();
+        this.modelDocTemplateFiles.clear();
+        this.modelTestTemplateFiles.clear();
+        this.operationTemplateFiles.clear();
+        this.supportingFiles.clear();
+
+        if (additionalProperties.containsKey(CodegenConstants.PACKAGE_NAME)) {
+            setPackageName((String) additionalProperties.get(CodegenConstants.PACKAGE_NAME));
+        }
+        else {
+            setPackageName("swagger");
+        }
+
+        if (additionalProperties.containsKey(CodegenConstants.PACKAGE_VERSION)) {
+            setPackageVersion((String) additionalProperties.get(CodegenConstants.PACKAGE_VERSION));
+        }
+        else {
+            setPackageVersion("1.0.0");
+        }
+
+        additionalProperties.put(CodegenConstants.PACKAGE_NAME, packageName);
+        additionalProperties.put(CodegenConstants.PACKAGE_VERSION, packageVersion);
+
+        apiPackage = packageName;
+
+        supportingFiles.add(new SupportingFile("README.mustache", "", "README.md"));
+        supportingFiles.add(new SupportingFile("Makefile.mustache", "", "Makefile"));
+        supportingFiles.add(new SupportingFile("gitignore.mustache", "", ".gitignore"));
+        supportingFiles.add(new SupportingFile("restclient.mustache", "/src/restclient", "restclient.go"));
+        apiTemplateFiles.put("api.mustache", ".go");
+    }
+
+    public String apiFileFolder() {
+        return (outputFolder + "/src/cmd").replace('/', File.separatorChar);
+    }
+
+    @Override
+    public String toApiFilename(String name) {
+        // replace - with _ e.g. created-at => created_at
+        name = name.replaceAll("-", "_"); // FIXME: a parameter should not be assigned. Also declare the methods parameters as 'final'.
+        String folder = name;
+
+        return toCustomApiName((folder + File.separatorChar + name));
+    }
+
+    @Override
+    public String toOperationId(String operationId) {
+        // method name cannot use reserved keyword, e.g. return
+        if (isReservedWord(operationId)) {
+            LOGGER.warn(operationId + " (reserved word) cannot be used as method name. Renamed to " + camelize(sanitizeName("call_" + operationId)));
```

Comment:
The value used in the trace message is different than what it assigns on the next line. The trace message is probably unnecessary (a lot of the tracing in this project is).

Reply:
Yeah I'll scrap that log message